apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -79,3 +217,292 @@ def _from_table(self, arrow_table: pa.Table) -> pa.SparseCOOTensor:
             ]
         ).T
         return pa.SparseCOOTensor.from_numpy(coo_data, coo_coords, shape=self.shape)
+
+
+SingleAxisTableIterReturn = Iterator[BlockwiseTableReadIterResult]
+
+
+def _single_axis_blockwise_table_iter(
+    array: "SparseNDArray",
+    sr: clib.SOMAArray,
+    coords: options.SparseNDCoords,
+    axis: int,
+    size: int,
+    reindex_disable: List[int],
+    eager: bool = True,
+    pool: Optional[concurrent.futures.Executor] = None,
+) -> SingleAxisTableIterReturn:
+    """Private. Blockwise Arrow Table iterator, restricted to a single axis"""
+
+    ndim = len(sr.shape)
+    if axis < 0 or axis >= ndim:
+        raise ValueError("chunk_dim argument must specify a dimension value.")
+
+    # normalize params tuple
+    coords = tuple(coords[i] if i < len(coords) else None for i in range(ndim))
+
+    # Materialize all joinids except the axis
+    joinids: List[pa.Array] = [
+        pa.array(
+            np.concatenate(list(_coords_strider(coords[d], sr.shape[d], sr.shape[d])))
+            if d != axis
+            else np.array([], dtype=np.int64)
+        )
+        for d in range(ndim)
+    ]
+
+    # build indexers, as needed
+    axes_to_reindex = set(range(ndim)) - set(reindex_disable)
+    minor_axes_indexer = {
+        d: pd.Index(joinids[d].to_numpy()) for d in (axes_to_reindex - set((axis,)))
+    }
+
+    def _if_eager(x: Iterator[RT]) -> Iterator[RT]:
+        return EagerIterator(x, pool=pool) if eager else x
+
+    # blockwise reader
+    def _table_reader() -> SingleAxisTableIterReturn:
+        kwargs: Dict[str, object] = {"result_order": sr.result_order}
+        for coord_chunk in _coords_strider(coords[axis], sr.shape[axis], size):
+            sr.reset(**kwargs)
+            step_coords = [*coords]
+            step_coords[axis] = coord_chunk
+            array._set_reader_coords(sr, step_coords)
+            yield pa.concat_tables(_arrow_table_reader(sr)), tuple(
+                joinids[0:axis] + [pa.array(coord_chunk)] + joinids[axis + 1 : ndim]
+            )
+
+    # with reindexing
+    def _reindexed_table_reader() -> SingleAxisTableIterReturn:
+        _tbl_iter = _table_reader() if eager else _if_eager(_table_reader())
+        for tbl, coords in _tbl_iter:
+            pytbl = {}
+            for d in range(ndim):
+                col = tbl.column(f"soma_dim_{d}")
+                if d in axes_to_reindex:
+                    if d == axis:
+                        col = pd.Index(coords[axis]).get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                    else:
+                        col = minor_axes_indexer[d].get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                pytbl[f"soma_dim_{d}"] = col
+            pytbl["soma_data"] = tbl.column("soma_data")
+            yield pa.Table.from_pydict(pytbl), coords
+
+    if eager and not pool:
+        with concurrent.futures.ThreadPoolExecutor() as _pool:
+            pool = _pool
```

Comment:
Changing out name bindings to backdoor pass new values into the function(s) below is a little sketchy to me—I would find it clearer to explicitly pass this in as a parameter.

Reply:
resolved by explicitly passing pool to generators